server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -983,130 +982,132 @@ public void writeWriteable(Writeable writeable) throws IOException {
-                writeCause = false;
-            } else if (throwable instanceof IllegalStateException) {
-                writeVInt(14);
-            } else if (throwable instanceof LockObtainFailedException) {
-                writeVInt(15);
-            } else if (throwable instanceof InterruptedException) {
-                writeVInt(16);
-                writeCause = false;
-            } else if (throwable instanceof IOException) {
-                writeVInt(17);
-            } else if (throwable instanceof EsRejectedExecutionException) {
-                writeVInt(18);
-                writeBoolean(((EsRejectedExecutionException) throwable).isExecutorShutdown());
-                writeCause = false;
             } else {
-                final ElasticsearchException ex;
-                if (throwable instanceof ElasticsearchException && ElasticsearchException.isRegistered(throwable.getClass(), version)) {
-                    ex = (ElasticsearchException) throwable;
-                } else {
-                    ex = new NotSerializableExceptionWrapper(throwable);
-                }
-                writeVInt(0);
-                writeVInt(ElasticsearchException.getId(ex.getClass()));
-                ex.writeTo(this);
-                return;
-            }
-            if (writeMessage) {
-                writeOptionalString(throwable.getMessage());
+                writeVInt(7);
             }
-            if (writeCause) {
-                writeException(rootException, throwable.getCause(), nestedLevel + 1);
+            writeOptionalString(fse.getFile());
+            writeOptionalString(fse.getOtherFile());
+            writeOptionalString(fse.getReason());
+            writeCause = false;
+        } else if (throwable instanceof IllegalStateException) {
+            writeVInt(14);
+        } else if (throwable instanceof LockObtainFailedException) {
+            writeVInt(15);
+        } else if (throwable instanceof InterruptedException) {
+            writeVInt(16);
+            writeCause = false;
+        } else if (throwable instanceof IOException) {
+            writeVInt(17);
+        } else if (throwable instanceof EsRejectedExecutionException eree) {
+            writeVInt(18);
+            writeBoolean(eree.isExecutorShutdown());
+            writeCause = false;
+        } else {
+            final ElasticsearchException ex;
+            if (throwable instanceof ElasticsearchException && ElasticsearchException.isRegistered(throwable.getClass(), version)) {
+                ex = (ElasticsearchException) throwable;
+            } else {
+                ex = new NotSerializableExceptionWrapper(throwable);
             }
-            ElasticsearchException.writeStackTraces(throwable, this, (o, t) -> o.writeException(rootException, t, nestedLevel + 1));
+            writeVInt(0);
+            writeVInt(ElasticsearchException.getId(ex.getClass()));
+            throwable = ex;
```

Comment:
This previously called `ex.write(this)` here, which lost the `nestedLevel` parameter, causing the bug this fix is for.